wapitiCore/attack/mod_brute_login_form.py

```diff
@@ -0,0 +1,184 @@
+
+    def set_timeout(self, timeout):
+        self.time_to_sleep = str(1 + int(timeout))
+
+    def has_password_field(self):
+        list = ['email','user','mail']
+        content = requests.get(self.current_request_url).content
+        inputs = BeautifulSoup(content, features="lxml").findAll('input')
+        for _input in inputs:
+            if (_input.attrs['type']).lower() == 'submit':
+                self.submit_var_name = _input.attrs['name']
+                self.submit_var_value = _input.attrs['name']
+
+            if any(ext in (_input.attrs['name']).lower() for ext in list):
+                # this condition is used to check that the field is not of type "password". For example : user_password
+                if (_input.attrs['type']).lower() != 'password':
+                    self.username_parameter_field.append(_input.attrs['name'])
+
+            if (_input.attrs['type']).lower() == 'password' and self.current_request_method == 'POST':
+                self.password_parameter.append(_input.attrs['name'])
+                return True
+
+        return False
+
+    def check_success_auth(self, content_response: str):
+        with open(path_join(self.CONFIG_DIR, self.PAYLOADS_SUCCESS), 'r', errors='ignore') as success_pattern_file:
+            for success_pattern in success_pattern_file:
+                if success_pattern.strip("\n") in content_response:
+                    return True
+        with open(path_join(self.CONFIG_DIR, self.PAYLOADS_FAIL), 'r', errors='ignore') as fail_pattern_file:
+            for fail_pattern in fail_pattern_file:
+                if fail_pattern in content_response:
+                    return False
+        return False
+
+    def get_username_list(self):
+        with open(path_join(self.CONFIG_DIR, self.PAYLOADS_FILE_USER), 'r', errors='ignore') as username_file:
+            data = username_file.readlines()
+            for username in data:
+                self.payloads_username.append(username.strip())
+
+    def get_password_list(self):
+        password_file = open(path_join(self.CONFIG_DIR, self.PAYLOADS_FILE), 'r', errors='ignore')
+        while True:
+            data = password_file.readline()
+            if not data:
+                break;
+
+            self.payloads_password[0]=data.strip()
+
+    def inject_username_payload(self, original_request, username_payload):
+        for params_list in original_request.post_params:
+            if self.username_parameter_field in params_list:
+                original_request.post_params[original_request.post_params.index(params_list)][1] = username_payload
+        return original_request
+
+    def attack(self):
+        global false_response
+        http_resources = self.persister.get_links(attack_module=self.name) if self.do_get else []
+        forms = self.persister.get_forms(attack_module=self.name) if self.do_post else []
+        timeouted = False
+
+
+
+        for original_request in chain(http_resources, forms):
```

Comment:
You probably do not need to iterate over GET requests

Reply:
fixed